src/main/java/app/FruitOperationStrategy.java

```diff
@@ -0,0 +1,21 @@
+package app;
+
+import app.service.Operation;
+import java.util.List;
+import java.util.Map;
+
+public class FruitOperationStrategy {
+    private Map<String, Operation> fruitOperations;
+
+    public FruitOperationStrategy(Map<String, Operation> fruitOperations) {
+        this.fruitOperations = fruitOperations;
+    }
+
```

Comment:
This class must contain only the logic of creating and initialization of the map. Else logic should be in different class

Reply:
This class contain only the logic getting the object of implementation of a specific operation. In the constructor, we only get an already filled out map for search.